addons/vote.py

```diff
@@ -5,37 +5,117 @@
+            await ctx.send(f"Your vote for {choice} has been succesfully registered!")
         with open("votes.json", "w") as votefile:
             json.dump(self.vote_list, votefile)
-        try:
-            await ctx.message.delete()
-        except:
-            pass
 
     @commands.guild_only()
     @commands.command()
-    async def vote(self, ctx, choice):
+    async def poll(self, ctx, command, link="", name="", *, options=""):
+        if command == "create" or command == "start":
+            if not self.poll_ongoing:
+                if not link or not name or not options:
+                    return await ctx.send(f"You didn't provide an argument. You should fix that. Inputted: link-{link}, name-{name}, options-{options}.")
+                vote_options = []
+                self.pollcfg["name"] = name
+                self.pollcfg["link"] = link
+                for option in options.split(" | "):
+                    vote_options.append(option)
+                self.pollcfg["options"] = vote_options
+                with open("poll.json", "w") as pollfile:
+                    json.dump(self.pollcfg, pollfile)
+                with open(name+"_votes.json","w") as votefile:
+                    json.dump(self.vote_list, votefile)
+                self.poll_ongoing = True
+                await ctx.send("Poll successfully created!")
+            else:
+                await ctx.send("There is poll ongoing already!")
+                
+        if command == "close" or command == "end":
+            if self.poll_ongoing:
+                # prob add evi final tally before finishing poll
+                if not os.path.isdir("Polls/"):
+                    os.mkdir("Polls")
+                with open("Polls/"+ self.pollcfg["name"]+".json", "w") as pollbackup:
+                    json.dump(self.pollcfg, pollbackup)
+                with open("Polls/"+ self.pollcfg["name"]+"_votes.json", "w") as votesbackup:
+                    json.dump(self.vote_list, votesbackup)
+                self.pollcfg = {}
+                self.vote_list = {}
+                with open("poll.json", "w") as pollfile:
+                    json.dump(self.pollcfg, pollfile)
+                self.poll_ongoing = False
+                await ctx.send("Poll successfully closed!")
+            else:
+                await ctx.send("There is no ongoing poll!")
+
+    @commands.guild_only()
+    @commands.command()
+    @commands.check(is_poll_ongoing)
+    async def vote(self, ctx, choice=""):
+        if not choice:
+            return await ctx.send("You forgot to make a choice! Use `$pollinfo` to find all choices, or use `$vote cancel` to clear your vote.")
         await self.queue.put((choice, ctx)) # tuple -> immutable plus easy get
         await self.process_vote()
 
+    @commands.guild_only()
+    @commands.command()
+    @commands.check(is_poll_ongoing)
```

Comment:
No check() that validates if user is a member of staff, which was agreed upon since this is a rather intensive command.